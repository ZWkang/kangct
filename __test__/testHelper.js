// import React from 'react';
import { mount, shallow, render } from 'enzyme';

function setUp(type, ...arg) {
  const [Component, props = {}, children = ''] = arg;
  if (~['mount', 'shallow', 'render'].indexOf(type)) {
    return global[type](<Component {...props}>{children}</Component>);
  }
}

export function AutoMath(Component, props = {}, children = '') {
  const wrapper = setUp('render', Component, props, children);
  expect(wrapper).toMatchSnapshot();
}

export function ComponentRenderWithoutError(Component, props = {}, children = '') {
  const wrapper = setUp('mount', Component, props, children);
  expect(() => {
    wrapper.setProps({});
    wrapper.unmount();
  }).not.toThrow();
}

export function childrenShouldBeProps(Component, children) {
  const wrapper = setUp('mount', Component, {}, children);

  expect(wrapper.get(0).props.children).toEqual(children);
}

export function setup(Component, props) {
  const wrapper = setUp(Component, props, null);
  return {
    wrapper,
    props
  };
}
