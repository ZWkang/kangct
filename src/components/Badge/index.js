import React, { Component } from 'react';
import PropTypes from 'prop-types';
import styled, { css } from 'styled-components';
import is from 'styled-is';

const Badge = styled.span`
  border-radius: 6px;
  font-size: 12px;
  margin-right: 8px;
  border-radius: 2px;
  text-transform: uppercase;
  text-decoration: none;
  text-align: center;
  padding: 0 6px;
  color: #000;
  border: 1px solid #ccc;
  background: #eee;
  ${is('solid')`
    color: #fff;
    border: 0;
    background: #000;
  `}
`;

Badge.propsTypes = {
  solid: PropTypes.bool
};

const BadgeContainer = styled.div`
  position: relative;
  display: inline-flex;
  width: auto;
  height: auto;
`;
const BadgeContent = styled.div`
  position: absolute;
  top: -9px;
  right: -18px;
  border-radius: 6px;
  background: red;
  color: #fff;
  padding: 0 6px;
  font-size: 12px;
`;
const Wrapper = ({ children, content }) => {
  return (
    <BadgeContainer>
      {children}
      <BadgeContent>{content}</BadgeContent>
    </BadgeContainer>
  );
};
Wrapper.propsTypes = {
  children: PropTypes.node,
  content: PropTypes.node
};

Badge.Wrapper = Wrapper;

export default Badge;
