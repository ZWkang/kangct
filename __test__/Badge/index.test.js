import Badge from '@app/Badge/index.js';
import { AutoMath, ComponentRenderWithoutError } from '../testHelper';
import 'jest-styled-components';

function createBadgeProps({ solid = false }) {
  return {
    solid
  };
}

function createBadgeWrapperProps({ content = '' }) {
  return { content };
}

describe('Badge', () => {
  describe('Badge rendering', () => {
    it('Badge rendering without error', () => {
      ComponentRenderWithoutError(Badge, createBadgeProps({}));
    });
    it('Badge snapshot', () => {
      AutoMath(Badge, createBadgeProps({}));
    });
    it('should contain Wrapper', () => {
      expect(Badge.Wrapper).not.toBeUndefined();
    });
  });
  describe('Badge Interactions', () => {
    it('render as <span>', () => {
      const badgewrapper = mount(<Badge>test</Badge>);
      expect(badgewrapper.find('span')).toHaveLength(1);
      expect(badgewrapper.find('div')).toHaveLength(0);
    });
    it('test props solid', () => {
      const badgewrapper = mount(<Badge>test</Badge>);

      expect(badgewrapper.find(Badge)).toHaveStyleRule('border', '1px solid #ccc');
      badgewrapper.setProps({ solid: true });
      expect(badgewrapper.find(Badge)).toHaveStyleRule('border', '0');
      expect(badgewrapper.find(Badge)).toHaveStyleRule('color', '#fff');
      expect(badgewrapper.find(Badge)).toHaveStyleRule('background', '#000');
    });
  });
  describe('Badge Lifecycle', () => {});
});

describe('Badge#Wrapper', () => {
  describe('Badge#Wrapper rendering', () => {
    it('Badge#Wrapper rendering without error', () => {
      ComponentRenderWithoutError(Badge.Wrapper, createBadgeProps({}));
    });
    it('Badge#Wrapper rendering without error', () => {
      //   ComponentRenderWithoutError(Badge.Wrapper, createBadgeProps({}));
      AutoMath(Badge.Wrapper, createBadgeProps({}));
    });
  });
  describe('Badge#Wrapper Interactions', () => {
    it('props content should be work', () => {
      const wrapper = mount(
        <Badge.Wrapper content="here is a test">
          <h1>test</h1>
        </Badge.Wrapper>
      );
      // console.log(wrapper.debug());
    });
  });
  describe('Badge#Wrapper Interactions', () => {});
});
