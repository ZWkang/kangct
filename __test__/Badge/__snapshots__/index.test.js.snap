// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Badge Badge rendering Badge snapshot 1`] = `
initialize {
  "0": Object {
    "attribs": Object {
      "class": "sc-bdVaJa IwViu",
    },
    "children": Array [],
    "name": "span",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": null,
    "prev": null,
    "root": Object {
      "attribs": Object {},
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "namespace": "http://www.w3.org/1999/xhtml",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
      "x-attribsNamespace": Object {},
      "x-attribsPrefix": Object {},
    },
    "type": "tag",
    "x-attribsNamespace": Object {
      "class": undefined,
    },
    "x-attribsPrefix": Object {
      "class": undefined,
    },
  },
  "_root": [Circular],
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "normalizeWhitespace": false,
    "withDomLvl1": true,
    "xml": false,
  },
}
`;

exports[`Badge#Wrapper Badge#Wrapper rendering Badge#Wrapper rendering without error 1`] = `
initialize {
  "0": Object {
    "attribs": Object {
      "class": "sc-bwzfXH mxVpw",
    },
    "children": Array [
      Object {
        "attribs": Object {
          "class": "sc-htpNat ksYogR",
        },
        "children": Array [],
        "name": "div",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": null,
        "parent": [Circular],
        "prev": null,
        "type": "tag",
        "x-attribsNamespace": Object {
          "class": undefined,
        },
        "x-attribsPrefix": Object {
          "class": undefined,
        },
      },
    ],
    "name": "div",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": null,
    "prev": null,
    "root": Object {
      "attribs": Object {},
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "namespace": "http://www.w3.org/1999/xhtml",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
      "x-attribsNamespace": Object {},
      "x-attribsPrefix": Object {},
    },
    "type": "tag",
    "x-attribsNamespace": Object {
      "class": undefined,
    },
    "x-attribsPrefix": Object {
      "class": undefined,
    },
  },
  "_root": [Circular],
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "normalizeWhitespace": false,
    "withDomLvl1": true,
    "xml": false,
  },
}
`;
